src/app/backend/client/manager.go

```diff
@@ -290,6 +291,47 @@ func (self *clientManager) extractTokenFromHeader(authHeader string) string {
 	return ""
 }
 
+func (self *clientManager) isLoginEnabled(req *restful.Request) bool {
+	return req.Request.TLS != nil || args.Holder.GetEnableInsecureLogin()
+}
+
+// Secure mode means that every request to Dashboard has to be authenticated and privileges
+// of Dashboard SA can not be used.
+func (self *clientManager) isSecureModeEnabled(req *restful.Request) bool {
```

Comment:
this logic is extremely difficult to follow. can we do something like this as a first step to make sure this always returns true when skip login is not enabled?

```
if !args.Holder.GetEnableSkipLogin() {
  return true
}
```

Reply:
It would have to be a bit different actually.

```go
if self.isLoginEnabled(req) && !args.Holder.GetEnableSkipLogin() {
  return true
}
```

In case login option is completely disabled we can not block user out of Dashboard. First we have to make sure that it is enabled and then skip option has to be disabled. Bascially if we would split that then whole method would become:
```go
func (self *clientManager) isSecureModeEnabled(req *restful.Request) bool {
	if self.isLoginEnabled(req) && !args.Holder.GetEnableSkipLogin() {
		return true
	}

	authInfo, _ := self.extractAuthInfo(req)
	return self.isLoginEnabled(req) && args.Holder.GetEnableSkipLogin() && authInfo != nil
}
```

It might be slightly easier to follow. @liggitt WDYT?

Reply:
that is simpler, but if I've configured the server to use TLS and not skip login, the incoming request shouldn't have any ability to make `isSecureModeEnabled` return `false`. Can we make the initial check only look at how the server was configured?

Reply:
> Can we make the initial check only look at how the server was configured, not look at req?

It would make this method even more complicated as we have several arguments responsible for TLS mode. The easiest way is to check if `req.Request.TLS != nil`. If it is then we are sure that the request itself is secure. 

There is no way to alter that from the outside. Only by corrupting `restful-go` used by dashboard someone would be able to somehow influence that.

Reply:
Letting a characteristic of an incoming request control bypassing security seems like a bad idea when the intent to serve securely can be determined from the config options alone. I'd like to see that adjusted in the future.